daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_functions/BitFunctions.tdml

```diff
@@ -0,0 +1,726 @@
+
+
+    <tdml:parserTestCase name="testUnsignedShortShifting" root="testUnsignedShortShifting" model="BitFunctions">
+      <tdml:document>1,15|65535,1|4,0|4,15</tdml:document>
+          <tdml:infoset>
+              <tdml:dfdlInfoset>
+                  <ex:testUnsignedShortShifting>
+                        <ex:leftUnsignedShort1>
+                            <num>1</num>
+                            <count>15</count>
+                            <value>32768</value>
+                        </ex:leftUnsignedShort1>
+                        <ex:leftUnsignedShort2>
+                            <num>65535</num>
+                            <count>1</count>
+                            <value>65534</value>
+                        </ex:leftUnsignedShort2>
+                        <ex:rightUnsignedShort1>
+                            <num>4</num>
+                            <count>0</count>
+                            <value>4</value>
+                        </ex:rightUnsignedShort1>
+                        <ex:rightUnsignedShort2>
+                            <num>4</num>
+                            <count>15</count>
+                            <value>0</value>
+                        </ex:rightUnsignedShort2>
+                  </ex:testUnsignedShortShifting>
+              </tdml:dfdlInfoset>
+          </tdml:infoset>
+    </tdml:parserTestCase>
+
+
+    <tdml:parserTestCase name="testUnsignedByteShifting" root="testUnsignedByteShifting" model="BitFunctions">
+      <tdml:document>1,7|255,1|4,0|4,7</tdml:document>
+          <tdml:infoset>
+              <tdml:dfdlInfoset>
+                  <ex:testUnsignedByteShifting>
+                        <ex:leftUnsignedByte1>
+                            <num>1</num>
+                            <count>7</count>
+                            <value>128</value>
+                        </ex:leftUnsignedByte1>
+                        <ex:leftUnsignedByte2>
+                            <num>255</num>
+                            <count>1</count>
+                            <value>254</value>
+                        </ex:leftUnsignedByte2>
+                        <ex:rightUnsignedByte1>
+                            <num>4</num>
+                            <count>0</count>
+                            <value>4</value>
+                        </ex:rightUnsignedByte1>
+                        <ex:rightUnsignedByte2>
+                            <num>4</num>
+                            <count>7</count>
+                            <value>0</value>
+                        </ex:rightUnsignedByte2>
+                  </ex:testUnsignedByteShifting>
+              </tdml:dfdlInfoset>
+          </tdml:infoset>
+    </tdml:parserTestCase>
+
+
+
+
+
+  <tdml:parserTestCase name="leftShiftFloatError01" root="leftShiftFloat" model="BitFunctions">
+    <tdml:document>1.0,1</tdml:document>
+      <tdml:errors>
+        <tdml:error>Unsupported function: dfdlx:shiftLeft</tdml:error>
```

Comment:
These unsupported functions aren't the right error messages. This means that Daffodil doesn't know what the `dfdlx:shiftLeft` function is. That's because the function is `dfdlx:leftShift`. You should be getting an error message that says the type must be a subtype of xs:unsignedLong or xs:long. Looks like the test schemas are just using shiftLeft instead of leftShift.

It's also usually a good practice to look for Schema Definitino Error in instead of some other kind of error.

So, I would reccomend your tdml:errors should look something like this:

```xml
<tdml:errors>
  <tdml:error>Schema Definition Error</tdml:error>
  <tdml:error>dfdlx:leftShift</tdml:error>
  <tdml:error>First argument</tdml:error>
  <tdml:error>xs:float</tdml:error>
</tdml:error>
```
This way we aren't too specific about the error message, but we just require that it's an SDE in the leftShift function, something is wrong with the first argument, and it's related to the first argument being a float. Something similar for the rest of these error functions.